### PR TITLE
Refactor ByteAmount checking args method

### PR DESCRIPTION
**Pull request by mycFelix**
This PR related #1612 and #1613 is trying to use `Preconditions.checkArgument` to check arguments in `ByteAmount` 

The key changes are following:

* use `Preconditions.checkArgument` method
* modify the condition expression

cc @billonahill

### CODE REVIEW

File: heron/common/src/java/com/twitter/heron/common/basics/ByteAmount.java [hunk 1]
```diff
@@ -130,10 +130,8 @@ public ByteAmount minus(ByteAmount other) {
    * @throws IllegalArgumentException if addition would exceed Long.MAX_VALUE
    */
   public ByteAmount plus(ByteAmount other) {
-    if (Long.MAX_VALUE - asBytes() < other.asBytes()) {
-      throw new IllegalArgumentException(String.format(
-          "Adding %s to %s would exceed Long.MAX_LONG", other, this));
-    }
+    Preconditions.checkArgument(Long.MAX_VALUE - asBytes() >= other.asBytes(), String.format(
+        "Subtracting %s from %s would overshoot Long.MIN_LONG", other, this));
```
Comment (billonahill) [inline heron/common/src/java/com/twitter/heron/common/basics/ByteAmount.java]: fix error message

File: heron/common/src/java/com/twitter/heron/common/basics/ByteAmount.java [hunk 2]
```diff
@@ -144,10 +142,8 @@ public ByteAmount plus(ByteAmount other) {
    * @throws IllegalArgumentException if multiplication would exceed Long.MAX_VALUE
    */
   public ByteAmount multiply(int factor) {
-    if (asBytes() > Long.MAX_VALUE / factor) {
-      throw new IllegalArgumentException(String.format(
-          "Multiplying %s by %d would exceed Long.MAX_LONG", this, factor));
-    }
+    Preconditions.checkArgument(asBytes() <= Long.MAX_VALUE / factor, String.format(
+        "Subtracting %s from %s would overshoot Long.MIN_LONG", this, factor));
```
Comment (billonahill) [inline heron/common/src/java/com/twitter/heron/common/basics/ByteAmount.java]: fix error message

File: heron/common/src/java/com/twitter/heron/common/basics/ByteAmount.java (hunk 1 above)
Comment (mycFelix) [inline heron/common/src/java/com/twitter/heron/common/basics/ByteAmount.java]: my carelessness

Comment (mycFelix) [inline heron/common/src/java/com/twitter/heron/common/basics/ByteAmount.java]: my carelessness

**Comment by mycFelix** [comment]
@billonahill - I've addressed all comments

**Comment by billonahill** [comment]
👍

**Comment by objmagic** [comment]
👍

**Pull request merged**